apis/sound_pressure/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+#![no_std]
+
+use core::{cell::Cell, convert::TryInto};
+use libtock_platform::{share, DefaultConfig, ErrorCode, Subscribe, Syscalls};
+
+pub struct SoundPressure<S: Syscalls>(S);
+
+impl<S: Syscalls> SoundPressure<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// Initiate a pressure measurement.
+    /// This function is used both for synchronous and asynchronous readings
+    pub fn read() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, READ_PRESSURE, 0, 0).to_result()
+    }
+
+    /// Register an events listener
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32,)>>,
```

Comment:
I think we did this wrong here. We need to be able to register a listener that the user can actually use outside this crate, meaning that the listener needs to have a function that is called when the upcall is fired. 

Something similar to the [`ButtonListener`](https://github.com/tock/libtock-rs/blob/cff868b1fb2641f2d1b0f3edaae827cbcfdadef3/apis/buttons/src/lib.rs#L124) should work.